### PR TITLE
Newsletters: add 110 (2020-08-12)

### PR DESCRIPTION
**Pull request by harding**
- [x] Update releases/RCs on Tuesday @harding 
- [x] Review club @jnewbery @jonatack 
- [x] `Bitcoin Core #18991` @jnewbery 
- [x] `Eclair #1499` @dongcarl 
- [x] Veriphi field report

### CODE REVIEW

File: _posts/en/newsletters/2020-08-12-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+%}
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [LND 0.11.0-beta.rc2][lnd 0.11.0-beta] is a release candidate for the
+  next major version of this project.  It will allows accepting [large
```
Comment (jonatack) [inline _posts/en/newsletters/2020-08-12-newsletter.md:87]: will allows -> allows or will allow

File: _posts/en/newsletters/2020-08-12-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1 repo],
+[Hardware Wallet Interface (HWI)][hwi], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #18991][] Cache responses to GETADDR to prevent topology leaks FIXME:jnewbery
+
+- [Bitcoin Core #19620][] prevents Bitcoin Core from re-downloading
+  unconfirmed segwit transactions that attempt to spend non-standard
+  UTXOs.  An non-standard UTXO is one that uses a currently discouraged
```
Comment (jonatack) [inline _posts/en/newsletters/2020-08-12-newsletter.md:104]: s/An/A/

**Comment by jonatack** [comment]
News110 review club (bip155): e64672470c0148a9ab3e42b6e57da436d5b7504d

**Comment by jnewbery** [comment]
I've pushed Jon's review club summary to this branch.

**Comment by jnewbery** [comment]
I've pushed my summary of bitcoin core 18991

File: _posts/en/newsletters/2020-08-12-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+## Bitcoin Core PR Review Club
+
+*In this monthly section, we summarize a recent [Bitcoin Core PR Review Club][]
+meeting, highlighting some of the important questions and answers.  Click on a
+question below to see a summary of the answer from the meeting.*
+
+[Implement ADDRv2 support (part of BIP155)][review club #19031] is a PR
+([#19031][Bitcoin Core #19031]) by Vasil Dimov that proposes a partial
+implementation of the [BIP155][] `addrv2` P2P message.
```
Comment (vasild) [inline _posts/en/newsletters/2020-08-12-newsletter.md:74]: This gives the impression that the PR partially implements `addrv2` whereas it fully implements `addrv2` (and partially BIP155, for full BIP155 we need TORv3, I2P, CJDNS implemented too, using the `addrv2` messages).

So, maybe reword to: ... proposes a partial implementation of BIP155 - the `addrv2` P2P message which would enable nodes to send and receive TORv3, IP2 and other larger addresses.

File: _posts/en/newsletters/2020-08-12-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+The discussion covered the current `addr` message in Bitcoin Core, the BIP155
+`addrv2` message, how to signal support for `addrv2` on the P2P network, and the
+necessity of `addrv2` for the Bitcoin network to upgrade to Tor v3. Tor v2
+deprecation begins mid-September!
+
+{% include functions/details-list.md
+  q0="How does Bitcoin Core handle network addresses?"
+  a0="Bitcoin Core currently sees all peer addresses as being more or less in a
+      [single][19031 single] IPv6-like space. IPv4 and Tor v2 addresses are
```
Comment (vasild) [inline _posts/en/newsletters/2020-08-12-newsletter.md:84]: What is this syntax `[single][19031 single]`? Where can I see how it is rendered? https://github.com/bitcoinops/bitcoinops.github.io/blob/8c021652cd78cdf190a566cc02b41b107a8ec2d0/_posts/en/newsletters/2020-08-12-newsletter.md looks bad.

File: _posts/en/newsletters/2020-08-12-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+      serialized into [16 bytes][19031 fake] encoded as “fake” IPv6 addresses
+      from specific IPv6 networks reserved for them. For instance, Tor v2
+      addresses are persisted using the [“onioncat” IPv6 range][19031 onioncat].
+      The 16-byte maximum length of IPv6 increasingly limits which endpoints
+      Bitcoin nodes can connect to."
+  a0link="https://bitcoincore.reviews/19031#l-35"
+
+  q1="How are peer addresses stored?"
+  a1="Addresses are saved in the `peers.dat` file in a 16-byte format, which
+      corresponds to the length of IPv6 and Tor v2 addresses; the smaller-sized
```
Comment (vasild) [inline _posts/en/newsletters/2020-08-12-newsletter.md:94]: `s/ and Tor v2//` (TORv2 addresses are 10 bytes)

File: _posts/en/newsletters/2020-08-12-newsletter.md (hunk 4 above)
Comment (jonatack) [reply _posts/en/newsletters/2020-08-12-newsletter.md:84]: (explained the link format via irc)

(you can render the document by building and running make preview per the readme and opening localhost:4000)

File: _posts/en/newsletters/2020-08-12-newsletter.md (hunk 5 above)
Comment (jonatack) [reply _posts/en/newsletters/2020-08-12-newsletter.md:94]: The review club study notes might need to be corrected: "Relevant to PR 19031 is the fact that Tor v2 addresses are 16 characters long, nicely fitting in the 16 bytes needed for an IPv6 address."

Comment (jonatack) [reply _posts/en/newsletters/2020-08-12-newsletter.md:94]: (Yes, there is the 6-byte onion prefix of `0xFD,0x87,0xD8,0x7E,0xEB,0x43`).

File: _posts/en/newsletters/2020-08-12-newsletter.md (hunk 3 above)
Comment (jonatack) [reply _posts/en/newsletters/2020-08-12-newsletter.md:74]: Maybe the shorter "partial implementation of BIP155." We go into the details further down.

File: _includes/articles/veriphi-segwit-batching.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+input isn’t native segwit, it calculates the weight it would save if it were
+native segwit, which can then indicate to us the amount of fees it could save.
+Finally, we collect the block weight and block fee of each block in reality and
+also collect the potential block weight and potential block fee of our model by
+summing up the results of each transaction. You can find the repository of our
+code used [here][veriphi segwit repo].
+
+__Major takeaways__
+
+Assuming a bitcoin price of $9250.00 (correct at the time of publication):
```
[the remark this answers is not in the document]
Comment (harding) [reply _includes/articles/veriphi-segwit-batching.md:41]: Yeah, but where will it be Wednesday morning?  :-D

File: _posts/en/newsletters/2020-08-12-newsletter.md (hunk 5 above)
Comment (harding) [reply _posts/en/newsletters/2020-08-12-newsletter.md:94]: Should this instead be `s/and Tor v2/and onioncat-encoded Tor v2/` ?

**Comment by harding** [comment]
Pushed edits for feedback.  Some notable unaddressed feedback:

- Update BTC→USD values to current values (more work than it's worth IMO) https://github.com/bitcoinops/bitcoinops.github.io/pull/449#discussion_r467691581
- Decrease precision of financial numbers (style decision IMO, up to report authors) https://github.com/bitcoinops/bitcoinops.github.io/pull/449#discussion_r467692759
- TorV2 storage in addrman (open question on what edit to make) https://github.com/bitcoinops/bitcoinops.github.io/pull/449#discussion_r468052888

If there are any other significant feedback that I've failed to address, please give me a kick.  Thanks everyone!

File: _posts/en/newsletters/2020-08-12-newsletter.md (hunk 5 above)
Comment (vasild) [reply _posts/en/newsletters/2020-08-12-newsletter.md:96]: Yes, "onioncat-encoded Tor v2" would be correct. Since we also store IPv4 in a similar way, maybe reword to: "which corresponds to the length of IPv6 and other addresses encoded as IPv6".

Comment (jonatack) [reply _posts/en/newsletters/2020-08-12-newsletter.md:96]: Within the context of this paragraph and the preceding ones, maybe:

"which corresponds to the length of IPv6 addresses; the smaller-sized IPv4 and Tor v2 addresses are therefore padded to fit."

Comment (jonatack) [reply _posts/en/newsletters/2020-08-12-newsletter.md:96]: (was tempted to link to `CNetAddr::SetLegacyIPv6`, but this code has changed considerably from 0.20 to master and the stable release code isn't as clear about it as it is now).

File: _posts/en/newsletters/2020-08-12-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+  from learning about all of the addresses in its address manager, Bitcoin Core
+  only provides a subset of those addresses when a peer sends it a `getaddr`
+  message, and only responds to a single `getaddr` message from each peer.
+  However, an adversary may still be able to learn all the addresses a peer knows
+  by making multiple connections to that peer and sending `getaddr` messages on
+  each connection. The adversary could potentially use that information to
+  fingerprint the peer or infer network topology. To prevent that, this PR
+  changes Bitcoin Core's behaviour to cache responses to `getaddr` requests and
+  provide the same response to all peers who request addresses in a 24 hour
+  window.
```
Comment (jonatack) [inline _posts/en/newsletters/2020-08-12-newsletter.md:169]: Great description @jnewbery! I wonder if it would be good to mention that it's a rolling window (randomly varying between 21 and 27 hours, if memory serves).

Comment (jnewbery) [reply _posts/en/newsletters/2020-08-12-newsletter.md]: good idea. Added the word "rolling"

File: _posts/en/newsletters/2020-08-12-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+  This helps address a concern described in [Newsletter #108][news108
+  wtxid] about a potential taproot activation: nodes that don't upgrade
+  for the soft fork won't accept unconfirmed taproot transactions, so
+  they might end up downloading and rejecting the same unconfirmed
+  taproot transactions over and over again.  That won't happen to any
+  node running this patch, allowing backports of this patch to serve as
+  an alternative to [backporting the wtxid relay feature][Bitcoin Core
+  #19606], which would also prevent that wasted bandwidth.
+
+- [C-Lightning #3909][] the `listpays` RPC now returns a new
```
Comment (jnewbery) [inline _posts/en/newsletters/2020-08-12-newsletter.md:185]: This isn't a complete sentence. Consider changing to "C-Lightning 3909 changes the `listpays` RPC to return..."

File: _posts/en/newsletters/2020-08-12-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+      [at handshake or afterwards][19031 handshake], rather than with the
+      originally proposed protocol version bump or with a new network service
+      bit."
+  a3link="https://bitcoincore.reviews/19031#l-99"
+
+  q4="Why is Tor v3 important for Bitcoin?"
+  a4="Tor v2 is planned to be [deprecated on Sept 15, 2020][19031 v2 deprecate]
+      and obsolete on July 15, 2021 ([schedule][19031 v2 schedule]). Tor v3
+      provides much better security and privacy. It uses longer, 56-character
+      addresses and therefore requires implementation of BIP155 `addrv2`."
```
[the remark this answers is not in the document]
Comment (jonatack) [reply _posts/en/newsletters/2020-08-12-newsletter.md:123]: > "Tor v3 provides much better security and privacy, since it uses longer, 56-character addresses. Using Tor v3 on the Bitcoin network therefore requires implementation of BIP155 `addrv2`."

Much better :+1:

**Comment by harding** [comment]
I pushed an edit to the Veriphi report reducing the precision of the the numbers as requested by @jonatack and @jnewbery (thanks!).  Although I do think that it's nicer that way, I don't have a strong opinion about it, so please feel free to revert.

**Comment by bitschmidty** [comment]
I conferred with Veriphi folks and they are ok with our judgment on these changes.

File: _includes/articles/veriphi-segwit-batching.md [hunk 10]
```diff
@@ -38,24 +38,21 @@ code used [here][veriphi segwit repo].
...
-  (Bech32). Fees would have amounted to 59,848.61 bitcoins, down from 96,534.33
-  bitcoins actually paid in fees, which is 38.00% in savings. The 36,685.72
-  bitcoins saved represent $339,342,910 USD.
+- Over 21 thousand BTC could have been saved by Bitcoin users if they
+  would have all been using transaction batching, a savings of 9.97% or
+  almost $200 million USD.
+- From 24 August 2017 to 30 June 2020, about 37 thousand BTC could have
+  been saved by Bitcoin users if they would have all been using segwit native
+  (bech32), down from over 97 thousand BTC
+  actually paid in fees, which is 38.00% in savings or about $340 million USD.
```
Comment (jonatack) [inline _includes/articles/veriphi-segwit-batching.md:55]: ACK on this commit 6067ddfac22b7dac. If you retouch, consider maybe changing 9.97% to 10% and 38.00% to 38%.

**Comment by jonatack** [comment]
ACK 40c7943e3abacff0a9ead1cd4cf20d4b763496d3 BCPRRC edits

**Comment by jnewbery** [comment]
ACK. Another smashing newsletter! Thanks team :hugs:

File: _posts/en/newsletters/2020-08-12-newsletter.md (hunk 5 above)
Comment (jonatack) [reply _posts/en/newsletters/2020-08-12-newsletter.md]: > The review club study notes might need to be corrected

done (for info)

**Comment by bitschmidty** [comment]
5 different authors this week 🙌 

Thanks to @harding @jonatack @jnewbery @dongcarl ! @vasild thanks for the review!
